Mathlib/Order/Chain.lean

```diff
@@ -136,6 +141,14 @@ theorem IsMaxChain.bot_mem [LE α] [OrderBot α] (h : IsMaxChain (· ≤ ·) s)
 theorem IsMaxChain.top_mem [LE α] [OrderTop α] (h : IsMaxChain (· ≤ ·) s) : ⊤ ∈ s :=
   (h.2 (h.1.insert fun _ _ _ => Or.inr le_top) <| subset_insert _ _).symm ▸ mem_insert _ _
 
+lemma IsMaxChain.image {s : β → β → Prop} (f : r ≃r s) {c : Set α} (hc : IsMaxChain r c) :
+    IsMaxChain s (f '' c) where
+  left := hc.isChain.image _ _ _ fun _ _ ↦ by exact f.map_rel_iff.2
+  right t ht hf := by
+    erw [← f.toEquiv.eq_preimage_iff_image_eq, preimage_equiv_eq_image_symm]
```

Comment:
Do you understand why this `erw` is needed? Can you avoid it?

Reply:
It is needed because the goal contains `⇑e` instead of the defeq `⇑e.toEquiv`. I avoided it by calling `RelIso.coe_fn_toEquiv` explicitly, but I'm not sure it's much better than just using `erw`